src/module/cache-container/CacheContainer.ts

```diff
@@ -54,6 +55,24 @@ module.config(($stateProvider: ng.ui.IStateProvider) => {
     }
   });
 
+  $stateProvider.state("container.counters", {
+    url: "/counters",
+    templateUrl: "module/cache-container/counters/view/counters.html",
+    controller: CountersCtrl,
+    controllerAs: "counterCtrl",
+    resolve: {
+      counters: ["container", "counterService", (container, counterService) => {
+        return counterService.doCountersExist(container.name, container.profile).then(countersExist => {
+          if (countersExist) {
+            return counterService.getAllCounters(container);
```

Comment:
Ideally `doCountersExist` would be part of `getAllCounters, with [] returned when counters don't exist, and that way the logic remains in the service.

Reply:
I did this